### ISSUE TITLE
Remove concept of Q (all tracers) from p3 hooks to the AD

### ISSUE THREAD

**Issue by bartgol** [E3SM-Project/scream#759]
This may have to wait till #758 is completed, but it should be done eventually.

As of now, having P3 accept a "packed" tracers array Q, and unpacking it internally, has a few drawbacks:

- atmosphere_microphysics.cpp needs to know qsize in order to size the Q array. It's mildly acceptable to have a SCREAM_QSIZE macro available (we need a compile time constant like that for Homme anyways).
- unpacking tracers might not be possible if the tracers layout is `(ncols,qsize,nlevs)`. We'd need them packed as `(qsize,ncols,nlevs)`.
- p3 is _assuming_ what each entry of Q represent. Even if the guesses are ok, all it takes is something being packed differently for the code to crash without any hint of what's wrong.

In short: the ad interface of P3 should assume all its required tracers are separate fields.

**Comment by AaronDonahue** [comment]
I'm happy to include the solution to this in the p3-interface code.  It seems like that will be a natural place to do it.

**Comment by bartgol** [comment]
Closing, cause I believe this is done already. @AaronDonahue , if I'm wrong, please, reopen.
